File: 0191-number-of-1-bits/python/Solution.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        oneCount = 0
        mask = 1
        for i in range(32):
            if n & mask != 0:
                oneCount += 1
            mask <<= 1
        return oneCount

class Solution:
    def hammingWeight(self, n: int) -> int:
        oneCount = 0
        i = 1
        while n != 0:
            if (n & 1) == 1:
                oneCount += 1
            n >>= 1
        return oneCount

File: 0191-number-of-1-bits/cpp/Solution.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int oneCount = 0;
        long mask = 1;
        for (int i = 0; i < 32; ++i) {
            if ((n & mask) != 0) {
                oneCount++;
            }
            mask <<= 1;
        }
        return oneCount;
    }
};

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int oneCount = 0;
        int i = 1;
        while (n != 0) {
            if ((n & 1) == 1) {
                oneCount++;
            }
            n >>= 1;
        }
        return oneCount;
    }
};
